src/collection/mod.rs

```diff
@@ -149,33 +149,55 @@ impl Collection {
                     let _ = self.insert_gauge(&metric.key(), *value as f64, &labels, None)?;
                 }
                 MetricType::Bytes(value) => {
-                    if self.options.exporter_skip_zero_metrics && value > &0 {
-                        let _ = self.insert_gauge(
-                            &metric.key(),
-                            *value as f64,
-                            &labels,
-                            Some("_bytes"),
-                        )?;
+                    if self.options.exporter_skip_zero_metrics && value == &0 {
+                        continue;
                     }
+                    let postfix = if metric.key().ends_with("_bytes") {
+                        None
+                    } else {
+                        Some("_bytes")
+                    };
+                    let _ = self.insert_gauge(&metric.key(), *value as f64, &labels, postfix)?;
                 }
                 MetricType::GaugeF(value) => {
-                    if self.options.exporter_skip_zero_metrics && value > &0.0 {
-                        let _ = self.insert_gauge(&metric.key(), *value, &labels, None)?;
+                    if self.options.exporter_skip_zero_metrics && value == &0.0 {
```

Comment:
I remember back from university that comparing floating point numbers like that might bring false positives. Can this be an issue?

Reply:
Rust handles this, added small test and re-factored to use `is_normal` that does the same but better